webdriver/tests/support/fixtures.py

```diff
@@ -38,6 +41,20 @@ def pytest_generate_tests(metafunc):
         if marker:
             metafunc.parametrize("capabilities", marker.args, ids=None)
 
+    if "bidi" in metafunc.fixturenames:
+        marker = metafunc.definition.get_closest_marker(name="bidi")
+        if marker:
+            metafunc.parametrize("bidi", marker.args, ids=None)
+
+# Reuse one event loop for all tests in the test session
+# so that we won't have the event loop closed issue when
```

Comment:
What's the "event loop closed issue"? (I assume it's soemthing like the event loop shutting down when there aren't any registered users, but it would be good to be clear).

Reply:
When I don't specify this, I get error such as
```
coro_or_future = <Task cancelled name='Task-10' coro=<WebSocketCommonProtocol.transfer_data() done, defined at /usr/local/google/home/shengfa/Workspace/wpt2/tools/third_party/websockets/src/websockets/protocol.py:818>>

    def ensure_future(coro_or_future, *, loop=None):
        """Wrap a coroutine or an awaitable in a future.

        If the argument is a Future, it is returned directly.
        """
        if coroutines.iscoroutine(coro_or_future):
            if loop is None:
                loop = events.get_event_loop()
            task = loop.create_task(coro_or_future)
            if task._source_traceback:
                del task._source_traceback[-1]
            return task
        elif futures.isfuture(coro_or_future):
            if loop is not None and loop is not futures._get_loop(coro_or_future):
>               raise ValueError('The future belongs to a different loop than '
                                 'the one specified as the loop argument')
E               ValueError: The future belongs to a different loop than the one specified as the loop argument
```
I found similar issue here: https://github.com/pytest-dev/pytest-asyncio/issues/171
I think it's because when I don't specify the scope to be session, it's by default function. Then the event loop will be closed after one test and restarted; hence the error "belongs to a different loop".

Reply:
That makes sense, thanks. Could you update the comment to explain this. Something like
```py
# Ensure that the event loop is restarted once per session rather than the default  of once per test
# if we don't do this, tests will try to reuse a closed event loop and fail with an error that the "future
# belongs to a different loop"
```

Reply:
Done, thanks for the suggestion!